src/main/java/org/folio/services/invoice/PoLinePaymentStatusUpdateService.java

```diff
@@ -214,14 +213,48 @@ private Future<Void> updatePoLinePaymentStatusUsingParameter(List<InvoiceLine> i
       return succeededFuture();
     }
     return orderLineService.getPoLinesByIdAndQuery(poLineIds, this::queryPoLinesWithOneTimeOpenOrder, requestContext)
-      .map(poLines -> {
-        poLines.forEach(poLine -> poLine.setPaymentStatus(PoLine.PaymentStatus.fromValue(poLinePaymentStatus)));
-        return poLines;
-      }).compose(poLines -> orderLineService.updatePoLines(poLines, requestContext))
+      .compose(poLines -> {
+        if (poLines.isEmpty()) {
+          logger.info("updatePoLinePaymentStatusUsingParameter:: No po line to update");
+          return succeededFuture();
+        }
+        poLines.forEach(poLine -> poLine.setPaymentStatus(PaymentStatus.fromValue(poLinePaymentStatus)));
+        return orderLineService.updatePoLines(poLines, requestContext);
+      })
       .onSuccess(v -> logger.info("updatePoLinePaymentStatusUsingParameter:: Success updating po lines"))
       .onFailure(t -> logger.error("updatePoLinePaymentStatusUsingParameter:: Error updating po lines", t));
   }
 
+  private Future<Void> updatePoLinePaymentStatusWithoutReleaseEncumbrance(List<InvoiceLine> invoiceLines, RequestContext requestContext) {
+    // This method is called when paying an invoice with the poLinePaymentStatus parameter,
+    // for invoice lines with releaseEncumbrance=false
```

Comment:
please also update docunentation for this parameter
https://github.com/folio-org/mod-invoice/blob/ef36ad90176825193b8e43815329b03f1c9fb464/ramls/invoice.raml#L72

Reply:
Done. But it's still true the parameter value only applies to lines with `releaseEncumbrance=true`.